DSharpPlus/DefaultLogger.cs

```diff
@@ -0,0 +1,195 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+
+namespace DSharpPlus
+{
+    public class DefaultLogger : ILogger<BaseDiscordClient>
```

Comment:
It contained one class initially. Ideally I'd spread it across multiple files, so that there's 1 class per file. Perhaps into its own folder (though not a namespace).